### ISSUE TITLE
April NodeSchool Workshoppers

### ISSUE THREAD

**Issue by jthoms1** [nodeschool/madison#3]
I am trying to figure out what workshoppers we should cover in the next session.  I think that scopes and closures cause a lot of problems for people who are new to JS.  Maybe we should cover this: [https://github.com/jesstelford/scope-chains-closures](https://github.com/jesstelford/scope-chains-closures)

I was also surpised that we still had a majority of people doing learnyounode in the february session.  Any thoughts/comments?

**Comment by davidpadbury** [comment]
I wouldn't be too surprised about most doing learnyounode. I think most I talked to would have only got through 3-4 exercises at the event and haven't worked on them during the interim. Unless we want to move to longer events or offline support, there's probably not much to be done.

Of the core workshops scope looks like a fine choice. A lot of the electives get pretty niche quickly (not that there's much wrong with that, just may not have much general appeal), then things like testing are technically useful but not particularly exciting.

Has there been any suggestions for content from attendees?

**Comment by jthoms1** [comment]
No suggestions, but I will probably ask in the next survey that I send out.  

@Pshrub and I talked today about maybe doing part of a session where we cover the solutions to Learnyounode and explain them. I will also pose this to attendees.

**Comment by briandennis** [comment]
I agree, most people I spoke to were still working on the learnyounode, I imagine that will be the case next time as well. Of the core ones, I think stream-adventure would be pretty cool/useful.

**Comment by jazeee** [comment]
I looked through the other exercises and I could see doing bug-clinic or test-anything.
I haven't run through them, but improving debugging skills or encouraging more testing seems like a good thing...
I could also see doing the scope-chains-closures or stream-adventure.  Both look interesting.

**Comment by jthoms1** [comment]
I think that debugging skills could be good.  I kind of want to go through the NPM and GIT. Having people use npm to publish their first package would be nice.

**Comment by jthoms1** [comment]
We should have everyone write an implementation of **left-pad** in a module, put it on github, and then publish to NPM.  Very timely and would probably be their first module published.

**Comment by jazeee** [comment]
Bonus points for including unit tests.

**Comment by jthoms1** [comment]
I was thinking I would supply the unit tests.

**Comment by jazeee** [comment]
That works.

**Comment by jthoms1** [comment]
ok, I will create a repo for it and start writing the docs tonight.  I also need to update the site and the meetup announcement.

**Comment by jthoms1** [comment]
http://nodeschool.io/madison/three/

**Comment by jazeee** [comment]
I've run through the workshoppers. A couple of notes:
1. git-it adds a collaborator called reporobot. When searching, github lists two matching accounts. Make sure to use reporobot meow (the first listing.)
2. how-to-npm asks to install linclark/pkg, which has changed to 1.0.3 recently. It seems to have messed up the test a little, while I ran through the workshopper. For example `npm outdated` did not work quite right, and blocked me from successfully completing `how-to-npm verify`.  I was able to complete it by renaming `.npmrc` to `.npmrc-local`. This allows `npm outdated` to hit npmjs.org and work as expected. `how-to-npm verify` completes successfully after that.
